sound/soc/sof/sof-priv.h

```diff
@@ -174,6 +174,7 @@ struct snd_sof_dsp_ops {
 	/* ipc */
 	int (*send_msg)(struct snd_sof_dev *sof_dev,
 			struct snd_sof_ipc_msg *msg); /* mandatory */
+	bool (*check_ipc_irq)(struct snd_sof_dev *sdev); /* optional */
```

Comment:
I know that were putting all callback ops under snd_sof_dsp_ops, but this one is really something which is HDA Intel / HDA only.
Should we have local Intel level ops for things that it is not a concern of any other architecture, platform?

Reply:
yes, we do something similar for SoundWire, there's no reason why this is needed at the top level. Fixed.